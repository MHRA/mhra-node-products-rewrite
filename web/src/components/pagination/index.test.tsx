import renderer from 'react-test-renderer';
import React from 'react';
import Pagination from '.';

const page = 2;
const pageSize = 10;
const resultCount = 50;
const searchTerm = 'Caffeine';

const dummyFunc = () => undefined;

describe(Pagination, () => {
  it('should render', () => {
    const component = renderer
      .create(
        <Pagination
          currentPage={page}
          pageSize={pageSize}
          resultCount={resultCount}
          searchTerm={searchTerm}
          handlePageChange={dummyFunc}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
