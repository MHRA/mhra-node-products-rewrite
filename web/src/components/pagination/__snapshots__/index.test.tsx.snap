// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render 1`] = `
<nav
  className="sc-bczRLJ wtnsX"
>
  <nav
    aria-label="Pagination Navigation"
  >
    <div
      className="pagination"
    >
      <button
        aria-label="Goto previous page, Page 1"
        className="arrow"
        onClick={[Function]}
      >
        Previous
      </button>
      <ul
        className="pagination-number"
      >
        <li
          aria-label="Goto Page 1"
          onClick={[Function]}
        >
          <span
            className="link-text"
          >
            1
          </span>
          <span>
            –
          </span>
        </li>
        <li
          aria-current="true"
          aria-label="Current Page, Page 2"
        >
          2
          <span>
            –
          </span>
        </li>
        <li
          aria-label="Goto Page 3"
          onClick={[Function]}
        >
          <span
            className="link-text"
          >
            3
          </span>
        </li>
        <li>
          …
        </li>
        <li
          aria-label="Goto Page 6"
          onClick={[Function]}
        >
          <span
            className="link-text"
          >
            6
          </span>
        </li>
      </ul>
      <button
        aria-label="Goto next page, Page 3"
        className="arrow"
        onClick={[Function]}
      >
        Next
      </button>
    </div>
  </nav>
</nav>
`;
